pkg/ruler/ruler.go

```diff
@@ -37,16 +37,16 @@ import (
 )
 
 var (
-	evalDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+	evalDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Namespace: "cortex",
 		Name:      "group_evaluation_duration_seconds",
 		Help:      "The duration for a rule group to execute.",
-	})
-	rulesProcessed = prometheus.NewCounter(prometheus.CounterOpts{
+	}, []string{"user_id", "filename"})
```

Comment:
Please don't put user_id in metrics. Cardinality gets too high too quickly.

Reply:
It can/should go in the trace span

Reply:
Ah, gotcha! Good point. I pushed for the histogram update, just because the granularity seems useful as clusters grow, so I'll take credit for that particular choice. Hadn't thought about adding it to the trace span!

Thanks a bunch for the input!

Reply:
Good to know! That change was in a second commit, removed now.